README.md

```diff
@@ -0,0 +1,175 @@
+# Open Context for Node.js
+
+This is a simple library for providing Distributed Context Problem
+in Node.js programs.
+
+DCP is most beneficial in large service oriented architectures. Each request
+has some associated metadata, and we want to make sure that data stays with
+the tree of RPCs that service the root request.
+
+The `Context` object in this library stores that request metadata, or baggage.
+It's intended to be implemented into RPC libraries, who would be responsible
+for actually serializing the data onto the wire.
+
+It also supports 'joining', a process where we copy all values from one
+context to another.
+
+A `Context` object is essentially just a bag of key/value pairs. By having a
+common interface, we can support `Context` in different RPC libraries.
+
+## `Context()`
+
+The `Context` object stores arbitrary key/value pairs. The constructor creates
+a new blank context.
+
+### `Context#set(key, value)`
+
+Maps `key` to `value` within this context.
+
+### `Context#get(key)`
+
+Gets the value of `key`.
+
+### `Context#keys()`
+
+Gets the list of keys.
+
+### `Context#joinWith(context, mergeFns)`
```

Comment:
Oh, right. In Go, this is not so well-hidden because the context is immutable and it becomes the client’s responsibility to join the prior context with the returned context for parallel requests. For serial requests, the RPC library can handle it behind the scenes, but for the parallel case, the "prior" context may not be the same as the request context.

I suspect there will be similar cases, where the need to join will leak out of the RPC library if there’s an explicit child context that needs to merge back to its parent, but I don’t have a concrete idea.

For now, 👍